kani-compiler/src/main.rs

```diff
@@ -141,7 +144,13 @@ fn main() -> Result<(), &'static str> {
 struct KaniCallbacks {}
 
 /// Use default function implementations.
-impl Callbacks for KaniCallbacks {}
+impl Callbacks for KaniCallbacks {
+    fn config(&mut self, config: &mut Config) {
+        if matches!(config.opts.error_format, ErrorOutputType::Json { .. }) {
```

Comment:
Is `error_format` the right place to hook? I thought that was for compiler errors. Does rustc format panic like this?

Reply:
Yes, to the best of my knowledge, that is the best place that works today. That is the option that `rustc` exposes to generate messages in `json` format: https://doc.rust-lang.org/rustc/command-line-arguments.html#--error-format-control-how-errors-are-produced. Which is used by `cargo` when you select `json` message format: https://doc.rust-lang.org/cargo/commands/cargo-build.html#display-options.

I initially tried to invoke `report_ice` API from `rustc`, but that actually did not produce a `json` message while reporting the ICE. I actually think that is a `rustc` issue. When `rustc` crashes, it is not emitting the panic message in `json` format. This is probably because the panic hook is initialized before parsing the arguments.

Reply:
BTW, I created https://github.com/rust-lang/rust/issues/107484 to see what they way about the panic message format.